File: client/src/models/job.ts
import type { JobCategory, JobLevel, SourceTrustLevel } from './enums';
import type { Tag } from './tag';

export interface Job {
  id: string;
  title: string;
  description: string;
  companyId: string;
  company: string;
  companySlug?: string;
  companyLogoUrl?: string;
  companyBannerUrl?: string;
  companyWebsite?: string;
  companyLinkedInUrl?: string;
  companyIndustry?: string;
  companySize?: string;
  companyDescription?: string;
  location?: string;
  city?: string;
  category: JobCategory;
  level: JobLevel;
  isRemote: boolean;
  isActive: boolean;
  isArchived?: boolean;
  salaryMin?: number;
  salaryMax?: number;
  expiresAt?: string;
  externalUrl?: string;
  jobImageUrl?: string;
  aiGeneratedImageUrl?: string;
  sourceId: string;
  sourceName?: string;
  sourceLogoUrl?: string;
  sourceTrustLevel?: SourceTrustLevel;
  sourceTrustScore?: number;
  tags: Tag[];
  trendingBadges?: string[];
  displayTitle?: string;
  displayCompany?: string;
  displaySalary?: string;
  displayLocation?: string;
  displaySkills?: string[];
  displayTags?: string[];
  displaySummary?: string;
  createdAt: string;
  updatedAt?: string;
}

export interface JobQuery {
  search?: string;
  page?: number;
  pageSize?: number;
  sortBy?: string;
  sortOrder?: string;
  category?: JobCategory;
  city?: string;
  isRemote?: boolean;
  salaryMin?: number;
  tags?: string;
  companyId?: string;
  companySlug?: string;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

export interface CreateJobRequest {
  title: string;
  description: string;
  company: string;
  location?: string;
  city?: string;
  category: JobCategory;
  level: JobLevel;
  isRemote: boolean;
  salaryMin?: number;
  salaryMax?: number;
  sourceId: string;
  tagIds?: string[];
}

File: server/src/SwipeJobs.Domain/Entities/Job.cs
using SwipeJobs.Domain.Common;
using SwipeJobs.Domain.Enums;

namespace SwipeJobs.Domain.Entities;

public class Job : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;

    public string? DisplayTitle { get; set; }
    public string? DisplayCompany { get; set; }
    public string? DisplaySalary { get; set; }
    public string? DisplayLocation { get; set; }
    /// <summary>JSON array of up to 5 card skill labels.</summary>
    public string? DisplaySkillsJson { get; set; }
    public string? DisplaySummary { get; set; }
    public string? Location { get; set; }
    public string? City { get; set; }
    public decimal? SalaryMin { get; set; }
    public decimal? SalaryMax { get; set; }
    public JobCategory Category { get; set; }
    public JobLevel Level { get; set; } = JobLevel.NotApplicable;
    public bool IsRemote { get; set; }
    public bool IsActive { get; set; } = true;
    public bool IsArchived { get; set; }
    public JobLifecycleStatus LifecycleStatus { get; set; } = JobLifecycleStatus.Published;
    public DateTime? PostedAt { get; set; }
    public DateTime? ExpiresAt { get; set; }
    public Guid? CandidateJobId { get; set; }
    public Guid? ApprovedByUserId { get; set; }
    public Guid? PublishedByUserId { get; set; }
    public DateTime? ApprovedAt { get; set; }
    public DateTime? PublishedAt { get; set; }
    public string? ExternalUrl { get; set; }
    public string? JobImageUrl { get; set; }
    public string? AiGeneratedImageUrl { get; set; }
    /// <summary>SHA-256 fingerprint for deduplication during ingestion.</summary>
    public string? ContentFingerprint { get; set; }
    /// <summary>Stable external key from the originating source (e.g. Telegram message id).</summary>
    public string? ExternalSourceKey { get; set; }

    public Guid CompanyId { get; set; }
    public Company Company { get; set; } = null!;

    public Guid SourceId { get; set; }
    public Source Source { get; set; } = null!;

    public ICollection<JobTag> JobTags { get; set; } = new List<JobTag>();
    public ICollection<Application> Applications { get; set; } = new List<Application>();
    public ICollection<SavedJob> SavedByUsers { get; set; } = new List<SavedJob>();
}
